src/mame/bfm/bfm_blackbox.h

```diff
@@ -0,0 +1,299 @@
+// license:BSD-3-Clause
+// copyright-holders:SomeRandomGuyIdk
+/**********************************************************************
+
+	Bellfruit Black Box
+
+**********************************************************************/
+
+#ifndef MAME_INCLUDES_BFM_BLACKBOX_H
+#define MAME_INCLUDES_BFM_BLACKBOX_H
+
+#pragma once
+
+#include "fruitsamples.h"
+
+#include "cpu/m6800/m6800.h"
+#include "cpu/tms1000/tms1000.h"
+#include "machine/6821pia.h"
+#include "machine/em_reel.h"
+#include "machine/timer.h"
+#include "sound/beep.h"
+#include "sound/spkrdev.h"
+
+#define OUT_READ(name) uint8_t name##_r(offs_t offset) { name##_w(offset, 0); return 0; }
+#define OUT_READ_T(name) template <unsigned T> uint8_t name##_r(offs_t offset) { name##_w<T>(offset, 0); return 0; }
+#define OUTPUT(name) rw(FUNC(name##_r), FUNC(name##_w))
+#define OUTPUT_T(name, T) rw(FUNC(name##_r<T>), FUNC(name##_w<T>))
+
+class blackbox_base_state : public driver_device
+{
+public:
+	blackbox_base_state(const machine_config &mconfig, device_type type, const char *tag) :
+		driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu"),
+		m_ram(*this, "ram"),
+		m_chute_timer(*this, "chute_timer"),
+		m_pia(*this, "pia"),
+		m_in_1800(*this, "IN1800"),
+		m_in_1800_en(*this, "IN1800_%u", 1U),
+		m_in_2000(*this, "IN2000"),
+		m_in_perc(*this, "PERCENTAGE"),
+		m_lamps(*this, "lamp%u", 0U),
+		m_digits(*this, "digit%u", 0U),
+		m_test_led(*this, "test%u", 1U),
+		m_samples(*this, "samples"),
+		m_nvram(*this, "nvram", 0x40, ENDIANNESS_BIG)
+	{ }
+
+	void blackbox_base(machine_config &config);
```

Comment:
If this is an abstract class and abstract machine configuration, this stuff should be `protected`.

Reply:
Fixed.